clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -964,6 +952,83 @@ private void handleShareAcknowledgeFailure(Node fetchTarget,
         }
     }
 
+    private void handlePartitionError(ShareAcknowledgeResponseData.PartitionData partitionData,
+                                      Map<TopicPartition, Metadata.LeaderIdAndEpoch> partitionsWithUpdatedLeaderInfo,
+                                      AcknowledgeRequestState acknowledgeRequestState,
+                                      Errors partitionError,
+                                      TopicIdPartition tip,
+                                      AtomicBoolean shouldRetry) {
+        if (partitionError.exception() != null) {
+            handleErrorConditionally(partitionData, partitionsWithUpdatedLeaderInfo, acknowledgeRequestState, partitionError, tip, shouldRetry);
```

Comment:
I would in-line this method's body again I think. It's clearer to follow without another method call.